scalasql/core/src/Queryable.scala

```diff
@@ -48,6 +48,15 @@ trait Queryable[-Q, R] {
    */
   def isSingleRow(q: Q): Boolean
 
+  /**
+   * Whether this query expects a single row to be returned, if so we can assert on
+   * the number of rows and raise an error if 2+ rows are present. Returns
+   * None if 0 rows are present.
+   * @param q
+   * @return
+   */
+  def isSingleRowOption(q: Q): Boolean
```

Comment:
I wonder if instead of adding another method `isSingleRowOption`, we should change the approach towards a method named `cardinality` (or something like that), that returns an enum value among: `singleOptional`, `single` or `many`.    

I think the danger with the current approach is that we end up with two somewhat mutually exclusive cases:  `isSingleRow` and `isSingleRowOption` ... what if they are both set to return true?